src/blocks/mrc_steps.ts

```diff
@@ -0,0 +1,285 @@
+        this.removeInput('STEP_' + currentIndex, false);
+
+        // Create new inputs at correct position
+        const fieldFlydown = createStepFieldFlydown(stepName, true);
+        fieldFlydown.setValidator(this.mrcUpdateStepName.bind(this, j));
+
+        this.appendValueInput('CONDITION_' + j)
+          .appendField(fieldFlydown)
+          .setCheck('Boolean')
+          .appendField(Blockly.Msg.REPEAT_UNTIL);
+        this.appendStatementInput('STEP_' + j);
+
+        // Reconnect
+        if (conditionConnection) {
+          this.getInput('CONDITION_' + j)?.connection?.connect(conditionConnection);
+        }
+        if (stepConnection) {
+          this.getInput('STEP_' + j)?.connection?.connect(stepConnection);
+        }
+
+        delete currentStepMap[stepName];
+      } else if (currentIndex !== undefined) {
+        // Step is at correct position - just update the field
+        const conditionInput = this.getInput('CONDITION_' + j);
+        const field = conditionInput?.fieldRow[0];
+        if (field && field.getValue() !== stepName) {
+          field.setValue(stepName);
+        }
+        delete currentStepMap[stepName];
+      } else {
+        // Step doesn't exist - create it
+        const fieldFlydown = createStepFieldFlydown(stepName, true);
+        fieldFlydown.setValidator(this.mrcUpdateStepName.bind(this, j));
+
+        const conditionInput = this.appendValueInput('CONDITION_' + j)
+          .appendField(fieldFlydown)
+          .setCheck('Boolean')
+          .appendField(Blockly.Msg.REPEAT_UNTIL);
+        this.appendStatementInput('STEP_' + j);
+
+        // Add shadow True block to the new condition input
+        if (this.workspace) {
+          const shadowBlock = this.workspace.newBlock('logic_boolean') as Blockly.BlockSvg;
+          shadowBlock.setShadow(true);
+          shadowBlock.setFieldValue('TRUE', 'BOOL');
+          if (this.workspace.rendered){
+            shadowBlock.initSvg();
+            shadowBlock.render();
+          }
+          conditionInput.connection?.connect(shadowBlock.outputConnection!);
+        }
+      }
+    }
+
+    // Remove any leftover inputs (steps that were deleted)
+    for (const stepName in currentStepMap) {
+      const index = currentStepMap[stepName];
+      this.removeInput('CONDITION_' + index, false);
+      this.removeInput('STEP_' + index, false);
+    }
+  },
+  mrcGetStepNames: function (this: StepsBlock): string[] {
+    return this.mrcStepNames;
+  }
+};
+
+export const setup = function () {
+  Blockly.Blocks[BLOCK_NAME] = STEPS;
+};
+
+export function isStepsInWorkspace(workspace: Blockly.Workspace): boolean {
+  const blocks = workspace.getBlocksByType(BLOCK_NAME);
+  return blocks.length > 0;
+};
+
+export const pythonFromBlock = function (
+  block: StepsBlock,
+  generator: ExtendedPythonGenerator,
+) {
+  let code = 'def steps(self):\n';
+  code += generator.INDENT + 'if not self._initialized_steps:\n';
```

Comment:
The python code on this line should be `if not hasattr(self, "_initialized_steps"):`

Reply:
This is fixed in https://github.com/alan412/systemcore-blocks-interface/pull/17